plugins/none-ls/servers.nix

```diff
@@ -5,273 +5,48 @@
-    markdownlint_cli2 = pkgs.markdownlint-cli2;
-    markdownlint = pkgs.nodePackages.markdownlint-cli;
-    mix = pkgs.elixir;
-    nimpretty = pkgs.nim;
-    nixfmt = pkgs.nixfmt-classic;
-    nixpkgs_fmt = pkgs.nixpkgs-fmt;
-    opacheck = pkgs.open-policy-agent;
-    opentofu_fmt = pkgs.opentofu;
-    pg_format = pkgs.pgformatter;
-    phpcbf = pkgs.phpPackages.php-codesniffer;
-    phpcsfixer = pkgs.phpPackages.php-cs-fixer;
-    phpcs = pkgs.phpPackages.php-codesniffer;
-    prisma_format = pkgs.nodePackages.prisma;
-    ptop = pkgs.fpc;
-    puppet_lint = pkgs.puppet-lint;
-    qmlformat = pkgs.qt6.qtdeclarative;
-    qmllint = pkgs.qt6.qtdeclarative;
-    racket_fixw = pkgs.racket;
-    raco_fmt = pkgs.racket;
-    rego = pkgs.open-policy-agent;
-    rpmspec = pkgs.rpm;
-    sqlformat = pkgs.python3.pkgs.sqlparse;
-    staticcheck = pkgs.go-tools;
-    surface = pkgs.elixir;
-    swift_format = pkgs.swift-format;
-    teal = pkgs.luaPackages.tl;
-    terraform_fmt = pkgs.terraform;
-    terraform_validate = pkgs.terraform;
-    tidy = pkgs.html-tidy;
-    verible_verilog_format = pkgs.verible;
-    vint = pkgs.vim-vint;
-    write_good = pkgs.write-good;
-    xmllint = pkgs.libxml2;
+  inherit (import ./packages.nix pkgs) packaged unpackaged;
 
-    # Sources not present in nixpkgs
-    blade_formatter = null;
-    bsfmt = null;
-    bslint = null;
-    cljstyle = null;
-    cueimports = null;
-    erb_lint = null;
-    findent = null;
-    forge_fmt = null;
-    gccdiag = null;
-    gersemi = null;
-    markuplint = null;
-    mlint = null;
-    nginx_beautifier = null;
-    npm_groovy_lint = null;
-    ocdc = null;
-    packer = null;
-    perlimports = null;
-    pint = null;
-    pretty_php = null;
-    purs_tidy = null;
-    pyink = null;
-    reek = null;
-    regal = null;
-    remark = null;
-    rescript = null;
-    saltlint = null;
-    solhint = null;
-    spectral = null;
-    sqlfmt = null;
-    sql_formatter = null;
-    styler = null;
-    stylint = null;
-    swiftformat = null;
-    swiftlint = null;
-    textidote = null;
-    textlint = null;
-    twigcs = null;
-    vacuum = null;
-
-    # Sources without packages
-    gitrebase = false;
-    # TODO: Requires the go tree-sitter parser
-    gomodifytags = false;
-    # TODO: Requires the go tree-sitter parser
-    impl = false;
-    luasnip = false;
-    printenv = false;
-    refactoring = false;
-    spell = false;
-    tags = false;
-    todo_comments = false;
-    trail_space = false;
-    ts_node_action = false;
-    vsnip = false;
-  };
-
-  # Check if the package is set to `false` or not
-  hasBuiltinPackage =
-    source:
-    if builtins.hasAttr source builtinPackages then
-      !(builtins.isBool builtinPackages.${source})
-    else
-      true;
-
-  builtinPackage = source: builtinPackages.${source} or null;
+  # Does this builitin require a package ?
+  hasBuiltinPackage = source: lib.hasAttr source packaged || lib.elem source unpackaged;
```

Comment:
The name is a little confusing. Would `builtinNeedsPackage` or `requiresPackage` (etc) be better?

Reply:
Definitively! I tried finding a better name, but didn't have enough inspiration